File: JS/firstnoncons.js
//Function has an array of integers
//Function must return the first element of the array that is not consecutive
//[1, 2, 3, 5] - 5
//Run a for loop starting with 1 as the index. Then we will check to see if the value of the previous index plus 1 is not equal to the current index. If it isn't equal, then we return the value of the current index. Otherwise we return null as default



function firstNonConsecutive (arr) {
  for (let i = 1; i < arr.length; i++) {
    if (arr[i - 1] + 1 !== arr[i]) {
      return arr[i]
      
    }
  } return null
}

console.log(firstNonConsecutive([1, 2, 3, 4, 6, 7]))
console.log(firstNonConsecutive([22, 23, 26, 27, 28]))
console.log(firstNonConsecutive([10, 14, 15, 16, 17, 18]))

File: py/firstnoncons.py

#Function has an array of integers
#Function must return the first element of the array that is not consecutive
#[1, 2, 3, 5] - 5
#Assign i to the first index of the array. Run a for loop and check to see if the iterating variable is equal to i. If it is not, return the iterating variable and then add 1 to i. 


def first_non_consecutive(arr):
    i = arr[0]
    for x in arr:
        if x != i:
            return x
        i += 1
    return None

print(first_non_consecutive([1, 2, 3, 4, 6, 7]))
print(first_non_consecutive([22, 23, 26, 27, 28]))